contrib/isa_l/compression/source/compressor/igzip_compressor_impl.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "envoy/compression/compressor/compressor.h"
+
+#include "contrib/isa_l/compression/source/common/base.h"
+#include "isa-l.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Igzip {
+namespace Compressor {
+
+const uint64_t memory_sizes[15] = {
+    ISAL_DEF_LVL1_MIN,         ISAL_DEF_LVL1_SMALL,       ISAL_DEF_LVL1_MEDIUM,
+    ISAL_DEF_LVL1_LARGE,       ISAL_DEF_LVL1_EXTRA_LARGE, ISAL_DEF_LVL2_MIN,
+    ISAL_DEF_LVL2_SMALL,       ISAL_DEF_LVL2_MEDIUM,      ISAL_DEF_LVL2_LARGE,
+    ISAL_DEF_LVL2_EXTRA_LARGE, ISAL_DEF_LVL3_MIN,         ISAL_DEF_LVL3_SMALL,
+    ISAL_DEF_LVL3_MEDIUM,      ISAL_DEF_LVL3_LARGE,       ISAL_DEF_LVL3_EXTRA_LARGE};
+
+/**
+ * Implementation of compressor's interface.
+ */
+class IgzipCompressorImpl : public Common::Base, public Envoy::Compression::Compressor::Compressor {
+public:
+  IgzipCompressorImpl();
+
+  IgzipCompressorImpl(uint64_t chunk_size);
+
+  // Note that only three compression levels supported in isa-l
+  enum class CompressionLevel : int64_t {
+    Best = 3,
+    Level1 = 1,
```

Comment:
Why we still need Level1/2/3 since we have speed/standard/best here?

Reply:
Level1/2/3 is more suitable to isa-l, speed/standard/best looks like redundant and needed to be removed. Thanks!